src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE = 10;
+    private static final double MULTIPLIER = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || size < index) {
+            throw new ArrayListIndexOutOfBoundsException("No valid index: " + index);
+        }
```

Comment:
It is better to move in private method
```suggestion
        if (index < 0 || size < index) {
            throw new ArrayListIndexOutOfBoundsException("No valid index: " + index);
        }
```

Reply:
What are the reasons why you didn't fix it?